tripal/src/TripalField/TripalFormatterBase.php

```diff
@@ -3,26 +3,113 @@
 namespace Drupal\tripal\TripalField;
 
 use Drupal\Core\Field\FormatterBase;
+use Drupal\Core\Field\FieldItemListInterface;
 
 /**
  * Defines the Tripal field formatter base class.
  */
 abstract class TripalFormatterBase extends FormatterBase {
 
+  /**
+   * Maximum number of records that will be published, 0 means no limit.
+   *
+   * @var int
+   */
+  public int $max_delta = 100;
+
   /**
    * Santizies a property key.
    *
    * Property keys are often controlled vocabulary IDs, which is the IdSpace
    * and accession separated by a colon. The colon is not supported by the
    * storage backend and must be converted to an underscore. This
-   * function performs that task
+   * function performs that task.
    *
    * @param string $key
+   *   A property key e.g. "operation:2945"
    *
    * @return string
    *   A santizied string.
    */
   public function sanitizeKey($key) {
     return preg_replace('/[^\w]/', '_', $key);
   }
+
+  /**
+   * {@inheritdoc}
+   *
+   * Initializes value of max_delta.
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) {
+    $max_delta = \Drupal::config('tripal.settings')->get('tripal_entity_type.publish_global_max_delta');
+    // max_delta defaults to 100 if not defined.
+    if (is_null($max_delta) or (trim($max_delta) === '')) {
+      $max_delta = 100;
+    }
+    $this->max_delta = $max_delta;
+  }
+
+  /**
+   * Converts a list of multiple renderable items into a list.
+   *
+   * A single item is passed through unchanged.
+   *
+   * @param array $list
+   *   A list of renderable items.
+   *
+   * @return array
+   *   A render array. Will be empty if there were no items.
+   */
+  public function createListMarkup(array $list): array {
+    $elements = [];
+
+    // If only one element has been found, don't make into a list.
+    if (count($list) == 1) {
+      $elements = $list;
+    }
+
+    // If more than one value has been found, display all values in an
+    // unordered list.
+    elseif (count($list) > 1) {
+      $this->addMaxDeltaWarning($elements, $list);
+      $elements[] = [
+        '#theme' => 'item_list',
+        '#list_type' => 'ul',
+        '#items' => $list,
```

Comment:
I don't see anything anywhere that removes the extra element from the end added by publish?

Reply:
It's one line earlier, line 74
`      $this->addMaxDeltaWarning($elements, $list);`
if the limit is exceeded, it does an array_slice()